### ISSUE TITLE
Zipkin dependency graph can be incorrect when stored in elasticsearch

### ISSUE THREAD

**Issue by mclarke47** [openzipkin/zipkin-dependencies#64]
I noticed several dependencies in our production Zipkin which I knew for a fact didn't exist. After some digging I managed to pull [an example of one of these traces](https://gist.githubusercontent.com/mclarke47/f922c16a503b9d88990d1361a16f49e7/raw/0abb6a8b140f92d01e5a7883e5d87ce6796d2f33/anon-%2520trace) out (and anonymised it).

I put these spans into a completely empty local in-memory Zipkin server and didn't see this incorrect link. However, when I set up this Zipkin server to use Elasticsearch and ran the dependency generator job, the incorrect dependency link was now present.

The trace 
<img width="1216" alt="screen shot 2017-04-05 at 15 24 03" src="https://cloud.githubusercontent.com/assets/6514980/24710373/de7d1c7a-1a14-11e7-87ae-6030dc6ceb98.png">

In memory storage
<img width="1278" alt="screen shot 2017-04-05 at 15 23 50" src="https://cloud.githubusercontent.com/assets/6514980/24710377/de84083c-1a14-11e7-80f7-673a5c47c754.png">

Elasticsearch storage
<img width="1274" alt="screen shot 2017-04-05 at 15 22 16" src="https://cloud.githubusercontent.com/assets/6514980/24710376/de7fc39e-1a14-11e7-8533-710246fde81b.png">

With a little direction I would gladly submit a PR to fix this

**Comment by adriancole** [comment]
is it possible for you to set ZIPKIN_LOG_LEVEL=DEBUG on the elasticsearch
job and send anonymized output of it (should already be pretty anonymous
except service names)

**Comment by adriancole** [comment]
we'd only need output for the trace in question

**Comment by mclarke47** [comment]
```
17/04/05 16:25:57 DEBUG DependencyLinker: processing {"traceId": "ecc61ea32e4938c7816dfac990274363", "parentId": "816dfac990274363", "id": "839d2c67ce7af4fc", "kind": "SERVER", "service": "service-a"}
17/04/05 16:25:57 DEBUG DependencyLinker: root's peer is unknown; skipping
17/04/05 16:25:57 DEBUG DependencyLinker: processing {"traceId": "ecc61ea32e4938c7816dfac990274363", "parentId": "839d2c67ce7af4fc", "id": "c652abfc8613951d", "kind": "UNKNOWN"}
17/04/05 16:25:57 DEBUG DependencyLinker: non-rpc span; skipping
17/04/05 16:25:57 DEBUG DependencyLinker: processing {"traceId": "ecc61ea32e4938c7816dfac990274363", "parentId": "816dfac990274363", "id": "ffb91320416b0c9b", "kind": "SERVER", "service": "service-b"}
17/04/05 16:25:57 DEBUG DependencyLinker: cannot determine parent, looking for first server ancestor
17/04/05 16:25:57 DEBUG DependencyLinker: processing ancestor {"traceId": "ecc61ea32e4938c7816dfac990274363", "parentId": "816dfac990274363", "id": "839d2c67ce7af4fc", "kind": "SERVER", "service": "service-a"}
17/04/05 16:25:57 DEBUG DependencyLinker: incrementing link service-a -> service-b
17/04/05 16:25:57 DEBUG DependencyLinker: processing {"traceId": "ecc61ea32e4938c7816dfac990274363", "parentId": "ffb91320416b0c9b", "id": "141d7016eebae1da", "kind": "UNKNOWN"}
17/04/05 16:25:57 DEBUG DependencyLinker: non-rpc span; skipping
17/04/05 16:25:58 INFO ElasticsearchDependenciesJob: Done
```

**Comment by adriancole** [comment]
thx assuming no-one beats me to it, I'll look at this in the morning MYT

(nag me if I forget)

**Comment by adriancole** [comment]
self-pinged.. looking into it

**Comment by adriancole** [comment]
ok this is not an issue with latest code, but.. build is broke :) I will ping when unbroke and release a version

**Comment by adriancole** [comment]
should be fixed in 1.5.2 lemme know if not
